### PR TITLE
Add Account management endpoints (CRUD)

### LINKED ISSUE #23: Add Account management endpoints
## Summary
Add account management endpoints. These require elevated permissions (Super Admin role, user-level API key).

## Endpoints to add

| Method | Path | SDK Method |
|--------|------|------------|
| `GET` | `/accounts` | `listAccounts()` |
| `POST` | `/accounts` | `createAccount()` |
| `GET` | `/accounts/{accountId}` | `getAccount()` |
| `DELETE` | `/accounts/{accountId}` | `deleteAccount()` |

## Notes
- `GET /accounts/{accountId}` supports `accountId: "show"` to get current account
- Create/delete require Super Admin role
- Delete queues deletion (not immediate)
- Supports `includes[]` query param (e.g., `sitoo_credentials`)
- These use user-level API keys, not account-level

## Acceptance criteria
- [ ] Methods added to `RuleClient`
- [ ] Types added to `src/types/api.ts`
- [ ] Exported from `src/index.ts`
- [ ] Tests added
- [ ] Document permission requirements in JSDoc

ref: [OpenAPI spec](https://app.rule.io/redoc/v3)

### PR DESCRIPTION
**Pull request by swesam**
## Summary
- Adds `listAccounts()`, `createAccount()`, `getAccount()`, and `deleteAccount()` to `RuleClient`
- `getAccount()` supports `'show'` param to get current account
- `listAccounts()` supports `includes[]` query param (e.g., `sitoo_credentials`)
- JSDoc documents Super Admin role requirement for create/delete
- Delete queues deletion (not immediate)

## Test plan
- [x] 10 new tests covering happy paths, edge cases (empty includes, "show" param, 404 null), error cases
- [x] All 224 tests pass
- [x] Type-check clean

Closes #23

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -1632,6 +1636,114 @@ export class RuleClient {
...
+  async listAccounts(params?: RuleAccountListParams): Promise<RuleAccountListResponse> {
+    let qs = '';
+    if (params?.includes?.length) {
+      const parts = params.includes.map(
+        (inc) => `includes[]=${encodeURIComponent(inc)}`
+      );
+      qs = '?' + parts.join('&');
+    }
+    return this.requestV3<RuleAccountListResponse>(`/accounts${qs}`, {
+      method: 'GET',
```
[the remark this answers is not in the document]
Comment (swesam) [reply src/client.ts]: Switched to `URLSearchParams` for the `includes[]` param to stay consistent with other array-param methods. Fixed in upcoming commit.
